src/main/java/sirius/web/dispatch/AssetsDispatcher.java

```diff
@@ -128,25 +134,46 @@ private DispatchDecision tryStaticResource(WebContext webContext, String uri, Re
         return DispatchDecision.DONE;
     }
 
-    private Tuple<String, Integer> getEffectiveURI(WebContext webContext) {
+    private void updateResponseCacheSettings(Response response, String uri, boolean constantAsset) {
+        if (uri.startsWith("/assets/dynamic/")) {
+            response.infinitelyCached();
+            return;
+        }
+
+        if (uri.startsWith("/assets/no-cache/")) {
+            response.notCached();
+            return;
+        }
+
+        if (constantAsset) {
+            response.cachedForSeconds((int) defaultStaticAssetTTL.getSeconds());
+            response.withDefaultCustomProxyTTL();
+            return;
+        }
+
+        response.cachedForSeconds(Response.obtainClientDurationInSeconds());
+    }
+
+    private String computeEffectiveURI(WebContext webContext) {
         String uri = webContext.getRequestedURI();
         if (uri.startsWith("/assets/dynamic/")) {
             uri = uri.substring(16);
```

Comment:
could we introduce a constant for the 16 which also documents what its for?
Same applies to 17

Reply:
Well, this is the length of the string just above. Its removing the prefix so /assets/dynamic/foo -> /foo
I think is straight forward, otherwise must put the string in a constant and get the .length of it